### ISSUE TITLE
[apex] ApexCRUDViolation doesn't report complex variable datatype operations inline DML

### ISSUE THREAD

**Issue by jonathanwiesel** [pmd/pmd#3378]
**Affects PMD Version:** 

5.5.3 - 6.36.0

**Rule:**

[ApexCRUDViolation](https://pmd.github.io/latest/pmd_rules_apex_security.html#apexcrudviolation)

**Description:**

When working with other datatypes beside a specific sObject or a List, inline operations in the DML statement fail to report the check

**Code Sample demonstrating the issue:**

```
public with sharing class CRUDCheck {
    
    public static void checkMe() {

        Map<String, Account> accountByName = new Map<String, Account>{'X' => new Account('X')};

        insert accountByName.values();
    }
}

```

**Expected outcome:**

PMD should report a violation at line 7, but doesn't. This is a false-negative.

**Running PMD through:** *Apex PMD VS Code Extension*
